### PR TITLE
Use recursive blob for package data

### PR DESCRIPTION
**Pull request by tzx**
setup.py now supports recursive glob for package data

I only added `.cpp`, `.h`, and `.yaml` files. Not sure if you want to include BAZEL or other files in package_data.

### CODE REVIEW

File: setup.py
Comment (zou3519) [inline setup.py:1336]: Is this feature limited only to certain versions of python?

Comment (tzx) [reply setup.py:1336]: This requires `setuptools` to have version >= 62.3.0, which implies needing a python version >= 3.7

**Comment by tzx** [comment]
@zou3519 Can you look at this again?

**Review by zou3519 — APPROVED** [review]

**Comment by zou3519** [comment]
Might need a rebase

**Comment by tzx** [comment]
@zou3519 oh I think I can't merge after a rebase right? Can I get this approved again?

**Review by zou3519 — APPROVED** [review]
LGTM as long as the tests pass

File: setup.py
Comment (malfet) [reply setup.py:1398]: Sorry, where did you find that statement?

**Comment by malfet** [comment]
Confirmed that indeed this change caused regression  by downloading wheels generated by https://github.com/pytorch/pytorch/commit/f20e3ae0c36146c962a5665018e9ad662a7cf211: https://github.com/pytorch/pytorch/actions/runs/8763218988  vs one generated by one commit before https://github.com/pytorch/pytorch/actions/runs/8763001691

```
 % unzip -t torch-2.4.0.dev20240420+cu118-cp38-cp38-linux_x86_64.whl|wc -l    
   11097
% unzip -t torch-2.4.0.dev20240420+cu118-cp38-cp38-linux_x86_64-bad.whl|wc -l
   11014
```

**Pull request closed without merging**
